StratisCore.UI/src/app/shared/services/staking-service.ts

```diff
@@ -0,0 +1,89 @@
+import { Injectable } from "@angular/core";
+import { BehaviorSubject, Observable } from "rxjs";
+import { StakingInfo } from "@shared/services/interfaces/api.i";
+import { catchError } from "rxjs/operators";
+import { RestApi } from "@shared/services/rest-api";
+import { GlobalService } from "@shared/services/global.service";
+import { HttpClient } from "@angular/common/http";
+import { ErrorService } from "@shared/services/error-service";
+import { SignalRService } from "@shared/services/signalr-service";
+import {
+  SignalREvents,
+  StakingInfoSignalREvent
+} from "@shared/services/interfaces/signalr-events.i";
+
+@Injectable({
+  providedIn: "root"
+})
+export class StakingService extends RestApi {
+  private stakingInfoUpdatedSubject = new BehaviorSubject<StakingInfo>(null);
+  public isStopping: boolean;
+  public isStarting: boolean;
+  public stakingEnabled: boolean;
+
+  constructor(
+    http: HttpClient,
+    globalService: GlobalService,
+    signalRService: SignalRService,
+    errorService: ErrorService) {
+    super(globalService, http, errorService);
+    signalRService.registerOnMessageEventHandler<StakingInfoSignalREvent>(
+      SignalREvents.StakingInfo, (message) => {
+        this.stakingInfoUpdatedSubject.next(message)
+      });
+  }
+
+  public startStaking(walletData: any): void {
+    this.isStarting = true;
+    this.isStopping = false;
+
+    this.invokeStartStakingApiCall(walletData)
+      .toPromise().then(
+      () => {
+        this.stakingEnabled = true;
+        this.isStarting = false;
+        return true;
+      },
+      error => {
+        console.log(error);
```

Comment:
Use `Log.debug()` or `Log.error` from logger.service, it does formatting and doesn't log for release mode.